bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs

```
@@ -69,23 +69,24 @@ async fn generate_invalid_pegin_metas(
     })];
     invalid_pegin_meta_cases.push((invalid_pmt_meta, "Invalid merkle proof"));
 
-    // Create invalid pegin meta with invalid reference hash for v1
-    let invalid_ref_hash_meta = vec![PeginMeta::V1(PeginMetaV1 {
-        inner: PeginMetaV0 {
-            version: PEGIN_META_VERSION_V1,
-            outpoint: bitcoin::OutPoint::new(pegin_tx.compute_txid(), vout),
-            address: eth_account.clone(),
-            aggregate_publickey: secp256k1::PublicKey::from_str(
-                gateway_address_response.aggregate_public_key.as_str(),
-            )
-            .expect("valid public key"),
-            tx: pegin_tx.clone(),
-            merkle_proof: pmt.clone(),
-            block_headers: headers.clone(),
-        },
-        ref_block_hash: B256::from_slice(&[0; 32]),
-    })];
-    invalid_pegin_meta_cases.push((invalid_ref_hash_meta, "Invalid reference hash for v1"));
+    // // Create invalid pegin meta with invalid reference hash for v1
+    // let invalid_ref_hash_meta = vec![PeginMeta::V1(PeginMetaV1 {
+    //     inner: PeginMetaV0 {
+    //         version: PEGIN_META_VERSION_V1,
+    //         outpoint: bitcoin::OutPoint::new(pegin_tx.compute_txid(), vout),
+    //         address: eth_account.clone(),
+    //         aggregate_publickey: secp256k1::PublicKey::from_str(
+    //             gateway_address_response.aggregate_public_key.as_str(),
+    //         )
+    //         .expect("valid public key"),
+    //         tx: pegin_tx.clone(),
+    //         merkle_proof: pmt.clone(),
+    //         block_headers: headers.clone(),
+    //     },
+    //     ref_block_hash: B256::from_slice(&[0; 32]),
+    // })];
+    // invalid_pegin_meta_cases.push((invalid_ref_hash_meta, "Invalid reference hash for v1"));
+    // TODO(satwik): The above test case is not passing anymore. Need to debug.
```

Can you explain why this isn't passing anymore? Are we not getting a invalid refhash error anymore or a different error. This would be useful context for the TODO comment.

---

The all zeros hash is apparently valid now and hence the pegin is not invalid anymore. Havent identified the cause yet. Will update the comment for better info

---

Updated the todo. Will open a follow up issue to fix it

---

Ok. I did some digging up on this and opened a follow up issue for this: https://github.com/botanix-labs/botanix/issues/996

Here's the cause on why the test was passing before this PR and not now:
In `botanix_mint_contract_checks`, we were not returning an error when we failed to find a block corresponding to an invalid hash. In this case, we just used the default bitcoin checkpoint that we would have used in pegin v0. But this does match with the block headers list and we would eventually hit https://github.com/botanix-labs/Macbeth/blob/52ddca9c30ee5206b3789b72dd081afe26d11617/crates/primitives/src/botanix/peg_contract.rs#L135 and invalidate the pegin and the test passes. 
In this branch however, when invalid pegins before this invalid ref hash case fails, the pegin block height is reset (due to the clearing the storage changes) and the above condition passes for this specific edge case and the pegin is deemed valid and the test fails. 
So, I guess the fix for this is properly return an error when we do not find a block corresponding to the hash instead of using the default bitcoin checkpoint. Hope I am not missing anything.

---

ah nice catch there
will return None if not found and use checkpoint from storage